src/packaging/specifiers.py

```diff
@@ -729,14 +722,8 @@ def __init__(
         # strip each item to remove leading/trailing whitespace.
         split_specifiers = [s.strip() for s in specifiers.split(",") if s.strip()]
 
-        # Parsed each individual specifier, attempting first to make it a
-        # Specifier.
-        parsed: Set[Specifier] = set()
-        for specifier in split_specifiers:
-            parsed.add(Specifier(specifier))
-
-        # Turn our parsed specifiers into a frozen set and save them for later.
-        self._specs = frozenset(parsed)
+        # Make each individual specifier a Specifier and save in a frozen set for later.
+        self._specs = frozenset(Specifier(specifier) for specifier in split_specifiers)
```

Comment:
```suggestion
        self._specs = frozenset(map(Specifier, split_specifiers))
```

I'd left this as a loop since it resulted in cleaner tracebacks (since the line did fewer things). If we decide to do this in one line, let's use `map`.